From 7d095a683dbd2ba71c434d246492136c3b97b92b Mon Sep 17 00:00:00 2001
From: Fengwei Yin <fengwei.yin@intel.com>
Date: Wed, 22 May 2013 12:35:34 +0800
Subject: [PORT FROM MAIN] stlport: Fix boundary condition in stlport for __get_integer.

BZ: 109103

The __get_integer doesn't trigger -min for signed (int, long, long long)
as overflow.

Change-Id: I6086a6671b082e8a06559d30e90c3708393cf356
Orig-Change-Id: I5140483774b2dec1a292532c7d44b5f2b2305554
Category: device enablement
Domain: AOSP-Bionic
Origin: internal
Upstream-Candidate: yes
---
 stlport/stl/_num_get.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/stlport/stl/_num_get.c b/stlport/stl/_num_get.c
index 1e7d234..65b918c 100644
--- a/stlport/stl/_num_get.c
+++ b/stlport/stl/_num_get.c
@@ -155,6 +155,9 @@ __get_integer(_InputIter& __first, _InputIter& __last,
      }
    }
 
+   if ((-__result == (numeric_limits<_Integer>::min)()) && !__is_negative)
+     __ovflow = true;
+
    if (__is_group && __group_sizes_end != __group_sizes) {
      *__group_sizes_end++ = __current_group_size;
    }
-- 
1.7.4.1

